Support 4.0 with key/value list

hanachin opened the pull request:
チケット: https://insidemine.agileware.jp/issues/19085

hanachin committed:
Fix NoMethodError: undefined method `before_filter' for ImporterController:Class

hanachin committed:
Fix NoMethodError: undefined method `parent=' for #<User:0x00007ffafb7434b8>

hanachin committed:
Fix ActiveRecord::RecordInvalid: Validation failed: Identifier cannot be blank, Identifier is too short (minimum is 1 characters)

hanachin committed:
Add .circleci/config.yml

hanachin committed:
Fix migration class

hanachin committed:
Remove attr_accessible

hanachin committed:
Fix: DEPRECATION WARNING: Using a dynamic :action segment in a route is deprecated and will be removed in Rails 6.0.

hanachin committed:
Fix delete_all

hanachin committed:
include ActiveJob::TestHelper to fix ActiveJob::DeserializationError

hanachin committed:
Support key value list

hanachin committed:
.presence is not needed, fix test instead

hanachin committed:
Fix controller test

hanachin committed:
Test against 4.0.4

hanachin committed:
Load Journal class before monkey patching

fkdkent commented on app/controllers/importer_controller.rb:
```diff
@@ -731,13 +734,33 @@ def version_id_for_name!(project, name, add_versions)
+      if enumeration.nil?
+        if name.try! { |n| n.size > 1 } && add_enumerations
+          max_position = custom_field.enumerations.pluck(:position).max || 0
+          enumeration = CustomFieldEnumeration.create(custom_field: custom_field, name: name, position: max_position + 1).id
```
> `position: max_position + 1`

Is this attribute needed? [`CustomFieldEnumeration#set_position`](https://github.com/redmine/redmine/blob/0454ec48a7d4e5cd45fa62177627ca698940fddd/app/models/custom_field_enumeration.rb#L78-L81) sets position before creation by default.

hanachin replied:
Thank you! I didn't know that. I'll fix it.

hanachin replied:
Fixed https://github.com/agileware-jp/redmine_importer/pull/5/commits/a6ac56038aafa4a29e7f20c99b5e978e8cd5b243 💪

hanachin committed:
Update orb

hanachin committed:
Remove unnecessary code

hanachin committed:
Remove add_enumeration checkbox

hanachin committed:
Do not send issue added notification when Send email notifications checkbox is unchecked

hanachin committed:
If user is not found, set nil to issue.assigned_to

hanachin committed:
If user is not found, set nil to user type custom field value

hanachin committed:
Update test

hanachin committed:
Clear Issue#assignet_to if is not assignable

fkdkent commented on test/functional/importer_controller_test.rb:
```diff
@@ -20,64 +22,164 @@ def setup
+    assert_equal @user, @issue.assigned_to
+  end
+
+  test 'should error when assigned_to user is not assignable' do
```
Will users see an error in this case? Maybe did you mean like this?

```suggestion
  test 'should unset assigned_to when assigned_to user is not assignable' do
```

hanachin replied:
That is correct, thank you 🙏

hanachin committed:
Update test/functional/importer_controller_test.rb

Co-Authored-By: fkdkent <fkdkent@agileware.jp>

hanachin committed:
ユーザーが存在するがユーザー型のカスタムフィールドの値として設定できないとき、エラーにせず空欄にする

Fix https://agileware.slack.com/archives/CF7CGT7FH/p1566441841001200?thread_ts=1566370267.014800&cid=CF7CGT7FH
